include/swift/AST/Decl.h

```diff
@@ -2207,6 +2207,16 @@ class ValueDecl : public Decl {
     return result;
   }
 
+  /// Determine whether this Decl has either Private or FilePrivate access.
+  bool isOutermostPrivateOrFilePrivateScope() const {
+    return ((hasAccess() && getFormalAccess() <= AccessLevel::FilePrivate) &&
+            !isInPrivateOrLocalContext());
+  }
+
+  /// Returns true if one of the ancestor DeclContexts of this ValueDecl is
+  /// either marked private or is a local context.
+  bool isInPrivateOrLocalContext() const;
```

Comment:
I'm not sure this is worth exposing to everyone. If it is, though, I think I'd rather put it on DeclContext.